### PR TITLE
update 404 and redirected links

### PR DESCRIPTION
**Pull request by eshaben**
### Description

This PR fixes links that are redirected to other pages to go straight to the new page now; of note, this includes all of the changes to the Moonbeam Network links and the Ethereum JSON RPC links.

Will do a separate check of the Chinese site after the slimmed down version is shipped.

### Checklist

- [x] I have added a label to this PR 🏷️

### CODE REVIEW

File: tutorials/eth-api/how-to-build-a-dapp.md [hunk 1]
```diff
@@ -1121,7 +1121,7 @@ Of course, there are more advanced (but optional) components of DApps that have
 - [Oracles](/builders/integrations/oracles/){target=\_blank} — third-party services that provide external data to smart contracts within blockchains
 - [Indexing protocols](/builders/integrations/indexers/){target=\_blank} — middleware that processes and organizes blockchain data, allowing it to be efficiently queried
 
-An excellent [Web2 to Web3 blogpost](https://moonbeam.network/blog/web2-vs-web3-development-heres-what-you-need-to-know-to-make-the-leap-to-blockchain){target=\_blank} is available if you are interested in hearing about them in depth.  
+An excellent [Web2 to Web3 blogpost](https://moonbeam.network/news/web2-vs-web3-development-here-s-what-you-need-to-know-to-make-the-leap-to-blockchain){target=\_blank} is available if you are interested in hearing about them in depth.
```
Comment (themacexpert) [inline tutorials/eth-api/how-to-build-a-dapp.md:1124]: This page 404s

File: tutorials/eth-api/chat-gpt.md [hunk 2]
```diff
@@ -134,7 +134,7 @@ If the problem is clear, ChatGPT will typically tell you exactly what's wrong an
...
 
-[A reentrancy bug](https://web.archive.org/web/20221121064906/https://consensys.github.io/smart-contract-best-practices/attacks/reentrancy){target=\_blank} was the root of the flaw that brought down the [original DAO on Ethereum in 2016](https://en.wikipedia.org/wiki/The_DAO_(organization)){target=\_blank}. Let's prompt ChatGPT with a buggy function that includes a reentrancy vulnerability and see if ChatGPT is able to spot the problem. We'll go ahead and copy and paste the below insecure code snippet into ChatGPT and ask if there is anything wrong with it.
+[A reentrancy bug](https://web.archive.org/web/20221121064906/https://consensys.github.io/smart-contract-best-practices/attacks/reentrancy/){target=\_blank} was the root of the flaw that brought down the [original DAO on Ethereum in 2016](https://en.wikipedia.org/wiki/The_DAO_(organization)){target=\_blank}. Let's prompt ChatGPT with a buggy function that includes a reentrancy vulnerability and see if ChatGPT is able to spot the problem. We'll go ahead and copy and paste the below insecure code snippet into ChatGPT and ask if there is anything wrong with it.
```
Comment (themacexpert) [inline tutorials/eth-api/chat-gpt.md:137]:
```suggestion
[A reentrancy bug](https://web.archive.org/web/20221121064906/https://consensys.github.io/smart-contract-best-practices/attacks/reentrancy/){target=\_blank} was the root of the flaw that brought down the [original DAO on Ethereum in 2016](https://en.wikipedia.org/wiki/The_DAO){target=\_blank}. Let's prompt ChatGPT with a buggy function that includes a reentrancy vulnerability and see if ChatGPT is able to spot the problem. We'll go ahead and copy and paste the below insecure code snippet into ChatGPT and ask if there is anything wrong with it.
```

File: tokens/connect/talisman.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ This guide will include all of the steps for setting up an account in Talisman a
...
+Talisman is a crypto-wallet that natively supports Substrate (Polkadot) and Ethereum accounts. The Talisman wallet browser extension is available on [Google Chrome](https://chromewebstore.google.com/detail/talisman-ethereum-and-pol/fijngjgcjhjmmpcmkeiomlglpeiijkld?pli=1){target=\_blank}, [Brave](https://chromewebstore.google.com/detail/talisman-ethereum-and-pol/fijngjgcjhjmmpcmkeiomlglpeiijkld){target=\_blank}, and [Firefox](https://addons.mozilla.org/en-US/firefox/addon/talisman-wallet-extension){target=\_blank}, and a corresponding asset dashboard is accessible at [app.talisman.xyz](https://app.talisman.xyz){target=\_blank}
```
Comment (themacexpert) [inline tokens/connect/talisman.md:20]:
```suggestion
Talisman is a crypto-wallet that natively supports Substrate (Polkadot) and Ethereum accounts. The Talisman wallet browser extension is available on [Google Chrome](https://chromewebstore.google.com/detail/talisman-ethereum-and-pol/fijngjgcjhjmmpcmkeiomlglpeiijkld){target=\_blank}, [Brave](https://chromewebstore.google.com/detail/talisman-ethereum-and-pol/fijngjgcjhjmmpcmkeiomlglpeiijkld){target=\_blank}, and [Firefox](https://addons.mozilla.org/en-US/firefox/addon/talisman-wallet-extension){target=\_blank}, and a corresponding asset dashboard is accessible at [app.talisman.xyz](https://app.talisman.xyz){target=\_blank}
```

File: builders/interoperability/xcm/xc20/send-xc20s/xtokens-pallet.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ description: This guide provides an introduction to the X-Tokens Pallet and expl
 
 ## Introduction {: #introduction }
 
-Building an XCM message for fungible asset transfers is not an easy task. Consequently, there are wrapper functions and pallets that developers can leverage to use XCM features on Polkadot and Kusama. One example of such wrappers is the [X-Tokens](https://github.com/moonbeam-foundation/open-runtime-module-library/tree/master/xtokens){target=\_blank} Pallet, which provides different methods to transfer fungible assets via XCM.
+Building an XCM message for fungible asset transfers is not an easy task. Consequently, there are wrapper functions and pallets that developers can leverage to use XCM features on Polkadot and Kusama. One example of such wrappers is the [X-Tokens](https://github.com/moonbeam-foundation/open-runtime-module-library/blob/master/xtokens){target=\_blank} Pallet, which provides different methods to transfer fungible assets via XCM.
```
Comment (themacexpert) [inline builders/interoperability/xcm/xc20/send-xc20s/xtokens-pallet.md:10]: When I go to the blob link it redirects to the tree link, i.e. the old link.

**Review by themacexpert — CHANGES_REQUESTED** [review]

File: tutorials/eth-api/how-to-build-a-dapp.md (hunk 1 above)
Comment (eshaben) [reply tutorials/eth-api/how-to-build-a-dapp.md:1124]: Really? Can you double-check it? It's working for me 🤔

File: builders/interoperability/xcm/xc20/send-xc20s/xtokens-pallet.md (hunk 4 above)
Comment (eshaben) [reply builders/interoperability/xcm/xc20/send-xc20s/xtokens-pallet.md]: good catch! thanks!

File: tutorials/eth-api/how-to-build-a-dapp.md (hunk 1 above)
Comment (themacexpert) [reply tutorials/eth-api/how-to-build-a-dapp.md:1124]: works!

**Review by themacexpert — APPROVED** [review]
LGTM 👍

**Pull request merged**